### PR TITLE
Fix portion clippy issues

### PR DESCRIPTION
**Pull request by alex-semenyuk**

### CODE REVIEW

File: src/config/file_lines.rs [hunk 1]
```diff
@@ -163,7 +163,7 @@ impl fmt::Display for FileLines {
             Some(map) => {
                 for (file_name, ranges) in map.iter() {
                     write!(f, "{}: ", file_name)?;
-                    write!(f, "{}\n", ranges.iter().format(", "))?;
+                    writeln!(f, "{}", ranges.iter().format(", "))?;
```
Comment (alex-semenyuk) [inline src/config/file_lines.rs:166]: warning: using `write!()` with a format string that ends in a single newline
```
   --> src\config\file_lines.rs:166:21
    |
166 |                     write!(f, "{}\n", ranges.iter().format(", "))?;
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
```
    = note: `#[warn(clippy::write_with_newline)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#write_with_newline

File: src/expr.rs [hunk 2]
```diff
@@ -78,12 +78,10 @@ pub(crate) fn format_expr(
         ast::ExprKind::Lit(ref l) => {
             if let Some(expr_rw) = rewrite_literal(context, l, shape) {
                 Some(expr_rw)
+            } else if let LitKind::StrRaw(_) = l.token.kind {
```
Comment (alex-semenyuk) [inline src/expr.rs:81]: warning: this `else { if .. }` block can be collapsed
  --> src\expr.rs:81:20
```
   |
81 |               } else {
   |  ____________________^
82 | |                 if let LitKind::StrRaw(_) = l.token.kind {
83 | |                     Some(context.snippet(l.span).trim().into())
84 | |                 } else {
85 | |                     None
86 | |                 }
87 | |             }
   | |_____________^
   |
```
   = note: `#[warn(clippy::collapsible_else_if)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_else_if

File: src/matches.rs [hunk 3]
```diff
@@ -427,12 +427,10 @@ fn rewrite_match_body(
                 };
                 let semicolon = if context.config.version() == Version::One {
                     ""
+                } else if semicolon_for_expr(context, body) {
```
Comment (alex-semenyuk) [inline src/matches.rs:430]: warning: this `else { if .. }` block can be collapsed
   --> src\matches.rs:430:24

File: src/patterns.rs [hunk 4]
```diff
@@ -328,14 +328,12 @@ fn rewrite_struct_pat(
...
             fields_str.push_str(&nested_shape.indent.to_string(context.config));
-        } else {
-            if !fields_str.is_empty() {
-                // there are preceding struct fields being matched on
-                if has_trailing_comma {
-                    fields_str.push(' ');
-                } else {
-                    fields_str.push_str(", ");
-                }
+        } else if !fields_str.is_empty() {
```
Comment (alex-semenyuk) [inline src/patterns.rs:331]: warning: this `else { if .. }` block can be collapsed
   --> src\patterns.rs:331:16

File: src/types.rs [hunk 5]
```diff
@@ -669,10 +669,12 @@ impl Rewrite for ast::Ty {
                 let shape = if is_dyn { shape.offset_left(4)? } else { shape };
                 let mut res = bounds.rewrite(context, shape)?;
                 // We may have falsely removed a trailing `+` inside macro call.
-                if context.inside_macro() && bounds.len() == 1 {
-                    if context.snippet(self.span).ends_with('+') && !res.ends_with('+') {
-                        res.push('+');
-                    }
+                if context.inside_macro()
```
Comment (alex-semenyuk) [inline src/types.rs:672]: warning: this `if` statement can be collapsed
   --> src\types.rs:672:17

**Comment by ytmimi** [comment]
@alex-semenyuk you might want to take a look at https://github.com/rust-lang/rustfmt/pull/5400 to make sure that you're not addressing the same lints.

**Comment by kadiwa4** [comment]
I was actually addressing everything that's enabled by default except `clippy::too_many_arguments`. It's nice that you're providing a little "changelog" in your PRs though :)

**Comment by alex-semenyuk** [comment]
Yeah, seems it was addressed at #5400

**Pull request closed without merging**